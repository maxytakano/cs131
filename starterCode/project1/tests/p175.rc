structdef STFU {
	int x;
	STFU* child;
};

function : void main() {
	
	int *x;
	int a;
	float *y;
	bool *z;
	bool res;
	STFU *s;
	
	res = x == a;
	res = a == x;
	res = x == y;
	res = s == *s;
}