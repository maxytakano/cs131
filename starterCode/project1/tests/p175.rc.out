Error, "/tmp/cs131_testrunner_gBTaux/file.rc", line 15: 
  Incompatible types to operator ==:
    int*,
    int;
  both must be of equivalent pointer type.
Error, "/tmp/cs131_testrunner_gBTaux/file.rc", line 16: 
  Incompatible types to operator ==:
    int,
    int*;
  both must be of equivalent pointer type.
Error, "/tmp/cs131_testrunner_gBTaux/file.rc", line 17: 
  Incompatible types to operator ==:
    int*,
    float*;
  both must be of equivalent pointer type.
Error, "/tmp/cs131_testrunner_gBTaux/file.rc", line 18: 
  Incompatible types to operator ==:
    STFU*,
    STFU;
  both must be of equivalent pointer type.
Compile: failure.
